tfhe/src/shortint/server_key/scalar_sub.rs

```diff
@@ -394,8 +398,12 @@ impl ServerKey {
     ///
     /// assert_eq!(msg - scalar as u64, clear);
     /// ```
+    #[allow(clippy::needless_pass_by_ref_mut)]
     pub fn smart_scalar_sub(&self, ct: &mut Ciphertext, scalar: u8) -> Ciphertext {
-        ShortintEngine::with_thread_local_mut(|engine| engine.smart_scalar_sub(self, ct, scalar))
+        let mut ct_result = ct.clone();
+        self.smart_scalar_sub_assign(&mut ct_result, scalar);
+
+        ct_result
```

Comment:
same thing do we want to update this function later to clean the input ct ?

did a follow up issue for those https://github.com/zama-ai/tfhe-rs-internal/issues/353